app/src/main/java/ayds/lisboa/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,204 @@
+package ayds.lisboa.songinfo.moredetails.fulllogic
+
+import androidx.appcompat.app.AppCompatActivity
+import android.widget.TextView
+import android.os.Bundle
+import ayds.lisboa.songinfo.R
+import retrofit2.Retrofit
+import retrofit2.converter.scalars.ScalarsConverterFactory
+import com.google.gson.Gson
+import com.google.gson.JsonObject
+import android.content.Intent
+import android.net.Uri
+import android.os.Build.*
+import com.squareup.picasso.Picasso
+import android.text.Html
+import android.text.Spanned
+import android.util.Log
+import android.view.View
+import android.widget.ImageView
+import retrofit2.Response
+import java.io.IOException
+import java.lang.StringBuilder
+import android.widget.Button
+
+const val ARTIST_NAME_EXTRA = "artistName"
+private const val LASTFM_API = "https://ws.audioscrobbler.com/2.0/"
+private const val IMAGE_URL_LASTFM = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/d4/Lastfm_logo.svg/320px-Lastfm_logo.svg.png"
+private const val ARTIST = "artist"
+private const val ARTIST_BIOGRAPHY = "bio"
+private const val ARTIST_BIOGRAPHY_EXTRACT = "content"
+private const val ARTIST_BIOGRAPHY_URL = "url"
+private const val ASTERISK = "[*]"
+private const val NO_RESULTS = "No Results"
+
+class OtherInfoWindow : AppCompatActivity() {
+
+    private lateinit var biographyTextView: TextView
+    private lateinit var openUrlButton: Button
+    private lateinit var imageView: ImageView
+    private lateinit var dataBase: DataBase
+    private lateinit var biographyJsonObject: JsonObject
+    private lateinit var serviceApi: LastFMAPI
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_other_info)
+
+        initProperties()
+        initLastFMAPI()
+        initDataBase()
+        getBiographyArtist()
+    }
+
+    private fun initProperties() {
+        biographyTextView = findViewById(R.id.biographyTextView)
+        openUrlButton = findViewById<View>(R.id.openUrlButton) as Button
+        imageView = findViewById<View>(R.id.imageView) as ImageView
+    }
+
+    private fun initLastFMAPI() {
+        serviceApi = initRetrofit().create(LastFMAPI::class.java)
+    }
+
+    private fun initRetrofit() = Retrofit.Builder()
+        .baseUrl(LASTFM_API)
+        .addConverterFactory(ScalarsConverterFactory.create())
+        .build()
+
+    private fun initDataBase(){
+        dataBase = DataBase(this)
+    }
+
+    private fun getBiographyArtist() {
+        val artist = getArtist()
+        getArtistInfo(artist)
+    }
+
+    private fun getArtist(): String? {
+        return intent.getStringExtra(ARTIST_NAME_EXTRA)
+    }
+
+    private fun getArtistInfo(artistName: String?) {
+        Thread {
+            updateArtistBiographyInDataBase(artistName)
+        }.start()
+    }
+
+    private fun updateArtistBiographyInDataBase(artistName: String?) {
```

Comment:
quedó mal este y otros nombres
```suggestion
    private fun getArtistBiographyAndUpdateUI(artistName: String?) {
```